selfservice/strategy/oidc/provider_generic_test.go

```diff
@@ -0,0 +1,46 @@
+package oidc
+
+import (
+	"context"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ory/kratos/selfservice/flow/login"
+	"github.com/ory/kratos/x"
+)
+
+func makeAuthCodeURL(t *testing.T, r *login.Request) string {
+	public, err := url.Parse("https://ory.sh")
+	require.NoError(t, err)
+	p := NewProviderGenericOIDC(&Configuration{
+		Provider:     "generic",
+		ID:           "valid",
+		ClientID:     "client",
+		ClientSecret: "secret",
+		IssuerURL:    "https://accounts.google.com",
+		SchemaURL:    "file://./stub/hydra.schema.json",
+	}, public)
+	c, err := p.OAuth2(context.TODO())
+	require.NoError(t, err)
+	return c.AuthCodeURL("state", p.AddAuthCodeURLOptions(r)...)
```

Comment:
I would rename this function to `p.AuthCodeURLOptions` because you are returning the options, not adding it to something.